### ISSUE TITLE
BigInt mapping in web with unexpected type

### ISSUE THREAD

**Issue by AlexandreAndrade00** [simolus3/moor#2961]
I have an application that uses the native and wasm Drift implementations. In the native version, the app works fine but when I run in the wasm implementation I'm getting an unexpected value in the drift mapping utilities.

![image](https://github.com/simolus3/drift/assets/46357905/ca3743a5-e097-4076-a39c-79e370064f3e)

Instead of getting an object with type _BigIntImp I'm getting a Map that appears to be a _BigIntImp converted.

Context:
The problem happens in the `id` column. 
```dart
class DynamicTables extends Table {
  Int64Column get id => int64().map(const TsidTypeConverter())();
  TextColumn get name => text()();
  TextColumn get permissions => text()();
  IntColumn get maxEntries => integer()();
  IntColumn get versions => integer()();

  @override
  Set<Column<Object>>? get primaryKey => {id};
}

class TsidTypeConverter extends TypeConverter<Tsid, BigInt> {
  const TsidTypeConverter();

  @override
  Tsid fromSql(BigInt fromDb) => Tsid.fromNumber(fromDb);

  @override
  BigInt toSql(Tsid value) => value.toLong();
}
```

I'm writing directly in the database with the executor:
```dart
// Got TransactionExecutor from Drift API from Database.executor.beginTransaction()
await transactionExecutor.runInsert("""
            INSERT INTO dynamic_tables (id, name, permissions, max_entries, versions)
            VALUES(?, ?, ?, ?, ?);
          """, [
          _mapper.mapToSqlLiteral(tableId.toLong()), // tableId is a Tsid, toLong() returns a BigInt
          _mapper.mapToSqlLiteral(name),
          _mapper.mapToSqlLiteral(''),
          _mapper.mapToSqlLiteral(-1),
          _mapper.mapToSqlLiteral(1),
        ]);

// I borrowed your mapper :)
String mapToSqlLiteral(Object? dart) {
    return switch (dart) {
      Null() => 'NULL',
      bool _ => dart ? '1' : '0',
      String() => dart.replaceAll('\'', '\'\''),
      num _ || BigInt _ => dart.toString(),
      DateTime() => (dart.millisecondsSinceEpoch ~/ 1000).toString(),
      Map<String, dynamic>() => json.encode(dart),
      _ => throw ArgumentError.value(dart, 'dart',
          'Must be null, bool, String, int, DateTime, Uint8List or double'),
    };
  }
```

**Comment by simolus3** [comment]
Thanks for the report! I think this is a bug in the worker that has since been fixed. Is it possible for you to upgrade to the [latest version](https://github.com/simolus3/drift/releases/tag/drift-2.17.0) and also use the latest worker? If that doesn't fix the issue I need to take another look, but I think the problem was that the big int is not getting properly serialized over the channel used to communicate with workers.

**Comment by AlexandreAndrade00** [comment]
The problem is resolved in the latest (2.17.0) version, thank you a lot!

**Comment by simolus3** [comment]
Glad to hear it's working, and thanks for the sponsorship :)

**Comment by AlexandreAndrade00** [comment]
Hello again @simolus3 , sorry for reopening the issue but I think the same error exists in batched statements. The operation with a BigInt as a parameter doesn't work when run in a batch in the web platform.

"Invalid argument (params[1]): Allowed parameters must either be null or bool, int, num, String or List<int>.: Instance of 'minified:H<dynamic>'"

When running: 
```dart
db.dynamicColumns.insertAll(List<DynamicColumns> aList);
```

```dart
class DynamicColumns extends Table {
  Int64Column get id => int64().map(const TsidTypeConverter())();
  TextColumn get name => text()();
  IntColumn get type => intEnum<DartSqlType>()();
  BoolColumn get system => boolean()();
  BoolColumn get isUnique => boolean()();
  BoolColumn get isNullable => boolean()();
  BoolColumn get allowsMultValues => boolean()();
  TextColumn get listingOptions => text().map(const ListConverter<String>())();
  Int64Column get foreignKeyId => int64()
      .unique()
      .nullable()
      .map(const TsidTypeConverter())
      .references(DynamicTableForeignKeys, #id)();
  Int64Column get dynamicTableId =>
      int64().map(const TsidTypeConverter()).references(DynamicTables, #id)();

  @override
  List<String> get customConstraints => [
        "CONSTRAINT column_unique_name_for_each_table UNIQUE (name, dynamic_table_id)"
      ];

  @override
  Set<Column<Object>>? get primaryKey => {id, dynamicTableId};
}
```

**Comment by simolus3** [comment]
Thanks for the follow-up! There's indeed a bug here, since `BigInt`s are encoded when sending batched statements but never properly decoded. I've fixed this in b1a5a11eef4d8d84c81e79bcf07c2fdb8c7ddaf1. Since only the decoding is affected you could try using an upgraded worker before the release (https://storage.googleapis.com/simon-public-euw3/assets/drift_worker.js) and see if that fixes it (I've only tested it with the latest changes on both ends but I think upgrading the worker is enough).
